devito/finite_differences/derivative.py

```diff
@@ -0,0 +1,162 @@
+import sympy
+
+from collections import OrderedDict
+from devito.finite_differences.finite_difference import (generic_derivative,
+                                                         first_derivative,
+                                                         cross_derivative)
+from devito.finite_differences.differentiable import Differentiable
+from devito.finite_differences.tools import centered, direct, transpose, left, right
+from devito.tools import as_tuple
+
+
+class Derivative(sympy.Derivative, Differentiable):
+
+    """
+    An unevaluated Derivative, which carries metadata (Dimensions,
+    derivative order, etc) to be evaluated.
+
+    Parameters
+    ----------
+
+    expr : symbolic expression
+    dims : Dimension or tuple
```

Comment:
this should be updated based on the inline comment below

Reply:
so, it should be something like

```
dims : Dimension or Iterable of Dimensions or ...
```

though in fact the actual elements could be 2-tuples?

Reply:
in the docstring you can use "tuple" instead of Iterable, but then the code it's better off to support all sort of tuple-like iterables